### PR TITLE
ci: run spread store tests when secret is available

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: .github/workflows/spread.yml [hunk 1]
```diff
@@ -68,3 +68,57 @@ jobs:
...
+
+      - if: steps.decisions.outputs.RUN == 'true'
+        name: Install spread
+        run: curl -s https://niemeyer.s3.amazonaws.com/spread-amd64.tar.gz | sudo tar xzv -C /usr/bin
+
+      - if: steps.decisions.outputs.RUN == 'true'
+        name: Run spread
+        env:
+          SPREAD_GOOGLE_KEY: ${{ secrets.SPREAD_GOOGLE_KEY }}
+          SNAP_STORE_MACAROON: ${{ secrects.SNAP_STORE_MACAROON }}
```
Comment (cjp256) [inline .github/workflows/spread.yml:115]:
```suggestion
          SNAP_STORE_MACAROON: ${{ secrets.SNAP_STORE_MACAROON }}
```

**Review by cjp256 — APPROVED** [review]
lgtm!

**Review by cmatsuoka — APPROVED** [review]
+1, but there's a failure with rosdep in the 18.04 tests.

**Review by Roadmaster — APPROVED** [review]
The variable settings in task.yaml look OK, confirmed they work as expected on my side.

The changes on staging_env.sh mirror that, so also +1 from me.

I've never touched github workflow yaml so no comment on those :)

**Pull request merged**